File: code_samples/typescript/oop_inheritance.ts
/**
 * Oop: Inheritance
 * AI/ML Training Sample
 */

interface IInheritance {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class Inheritance implements IInheritance {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new Inheritance();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { Inheritance, IInheritance };

File: code_samples/swift/oop_inheritance.swift
import Foundation

/// Oop: Inheritance
/// AI/ML Training Sample

class Inheritance {
    var data: String
    
    init() {
        self.data = ""
    }
    
    func process(_ input: String) {
        self.data = input
    }
    
    func getData() -> String {
        return self.data
    }
    
    func validate() -> Bool {
        return !self.data.isEmpty
    }
}

// Example usage
let instance = Inheritance()
instance.process("example")
print("Data: \(instance.getData())")
print("Valid: \(instance.validate())")
